src/ApiSchema.ts

```diff
@@ -129,78 +148,42 @@ export class APISchema {
         this.timeOut = setTimeout(this.updateSchema.bind(this), this.sharedSettings.apiUrlInterpreter.timeOutDuration);
     }
 
-    private constructBaseUrlRegex(schema: any): string {
-        const baseUrl = this.escapeRegex(schema.servers[0].url);
-        this.platforms = schema.servers[0].variables.platform.enum;
-
-        // This makes a regex string from all the platform options
-        this.platformRegexString = `(${this.platforms.join("|")})`;
-
-        return baseUrl.replace(/{platform}/g, this.platformRegexString);
+    /**
+     * Constructs a regex that will match an API endpoint and store each path parameter in a named group.
+     */
+    private constructRegex(path: string, methodSchema: any): RegExp {
+        // Allow an optional trailing slash
+        let regex = path + "/?";
+        // Escape slashes
+        regex = regex.replace(/\//g, "\\/");
+        // Replace each parameter with a named regex group
+        regex = regex.replace(/\{(.*?)\}/, (match, p2) => `(?<${p2}>[^\\/?\\s]*)`);
+
+        return XRegExp(regex);
     }
 
-    private constructRegex(path: Path, validBase: string, invalidBase: string, pathName: string, methodSchema: any) {
-
-        path.name = pathName;
-        path.parameterInfo = [];
+    private addParameter(map: Map<string, Parameter>, parameterSchema: any) {
+        const parameter = new Parameter();
+        parameter.name = parameterSchema.name;
+        parameter.required = parameterSchema.required;
+        parameter.type = this.getParameterType(parameterSchema.schema.type);
 
```

Comment:
Same goes with the class filling instead of a proper constructor here. Or you can just make Parameter an interface instead of a class and use plain objects